### PR TITLE
James 3621 get mail box size

### PR DESCRIPTION
**Pull request by vttranlina**
Ref: https://issues.apache.org/jira/browse/JAMES-3621

https://github.com/linagora/james-project/issues/4363

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java [hunk 1]
```diff
@@ -342,4 +346,42 @@ public void defineCreateUserMailbox() {
...
+
+    @GET
+    @Path("/{mailboxName}/messageCount")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = "username", paramType = "path"),
+        @ApiImplicitParam(required = true, dataType = "string", name = "mailboxName", paramType = "path")
+    })
+    @ApiOperation(value = "Counting emails in a given mailbox.")
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "The number emails in a given mailbox", response = String.class),
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java:358]: Isn't the bodu a `Long`?

**Comment by chibenwa** [comment]
We also miss the webadmin documentation here https://github.com/apache/james-project/blob/master/docs/modules/servers/pages/distributed/operate/webadmin.adoc#administrating-user-mailboxes and here https://github.com/apache/james-project/blob/master/src/site/markdown/server/manage-webadmin.md#administrating-user-mailboxes

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 2]
```diff
@@ -1322,6 +1322,38 @@ Warning: While we have been trying to reduce the inconsistency window to
 a maximum (by keeping track of ongoing events), concurrent changes done
 during the reIndexing might be ignored.
 
+=== Counting emails
+
+....
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/messageCount
+....
+
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:1330]: Will return the total count of messages within the mailbox of that user.

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 3]
```diff
@@ -1322,6 +1322,38 @@ Warning: While we have been trying to reduce the inconsistency window to
...
+* 200: The number emails in a given mailbox
+* 400: Invalid mailbox name
+* 404: Invalid get on user mailboxes. The `usernameToBeUsed` or `mailboxName` does not exit'
+
+=== Counting unseen emails
+
+....
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/unseenMessageCount
+....
+
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:1346]: Will return the total count of unseen messages within the mailbox of that user.

File: src/site/markdown/server/manage-webadmin.md [hunk 4]
```diff
@@ -1167,6 +1169,37 @@ Warning: Canceling this task should be considered unsafe as it will leave the cu
 Warning: While we have been trying to reduce the inconsistency window to a maximum (by keeping track of ongoing events),
 concurrent changes done during the reIndexing might be ignored.
 
+### Counting emails
+
+```bash
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/messageCount
+```
+
```
Comment (chibenwa) [inline src/site/markdown/server/manage-webadmin.md:1177]: Will return the total count of messages within the mailbox of that user.

File: src/site/markdown/server/manage-webadmin.md [hunk 5]
```diff
@@ -1167,6 +1169,37 @@ Warning: Canceling this task should be considered unsafe as it will leave the cu
...
+- 200: The number emails in a given mailbox
+- 400: Invalid mailbox name
+- 404: Invalid get on user mailboxes. The `usernameToBeUsed` or `mailboxName` does not exit'
+
+### Counting unseen emails
+
+```bash
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/unseenMessageCount
+```
+
```
Comment (chibenwa) [inline src/site/markdown/server/manage-webadmin.md:1192]: Will return the total count of unseen messages within the mailbox of that user.

**Review by chibenwa — APPROVED** [review]

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 6]
```diff
@@ -1322,6 +1322,38 @@ Warning: While we have been trying to reduce the inconsistency window to
...
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/messageCount
+....
+
+Resource name `usernameToBeUsed` should be an existing user.
+
+Resource name `mailboxName` should not be empty, nor contain `% *` characters, nor starting with `#`.
+
+Response codes:
+
+* 200: The number emails in a given mailbox
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:1339]:
```suggestion
* 200: The number of emails in a given mailbox
```

File: docs/modules/servers/pages/distributed/operate/webadmin.adoc [hunk 7]
```diff
@@ -1322,6 +1322,38 @@ Warning: While we have been trying to reduce the inconsistency window to
...
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/unseenMessageCount
+....
+
+Resource name `usernameToBeUsed` should be an existing user.
+
+Resource name `mailboxName` should not be empty, nor contain `% *` characters, nor starting with `#`.
+
+Response codes:
+
+* 200: The number unseen emails in a given mailbox
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/webadmin.adoc:1357]:
```suggestion
* 200: The number of unseen emails in a given mailbox
```

File: src/site/markdown/server/manage-webadmin.md [hunk 8]
```diff
@@ -1167,6 +1169,37 @@ Warning: Canceling this task should be considered unsafe as it will leave the cu
...
+```bash
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/messageCount
+```
+
+Resource name `usernameToBeUsed` should be an existing user.    
+Resource name `mailboxName` should not be empty, nor contain `% *` characters, nor starting with `#`.
+
+Response codes:
+
+- 200: The number emails in a given mailbox
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md:1185]:
```suggestion
- 200: The number of emails in a given mailbox
```

File: src/site/markdown/server/manage-webadmin.md [hunk 9]
```diff
@@ -1167,6 +1169,37 @@ Warning: Canceling this task should be considered unsafe as it will leave the cu
...
+```bash
+curl -XGET http://ip:port/users/{usernameToBeUsed}/mailboxes/{mailboxName}/unseenMessageCount
+```
+
+Resource name `usernameToBeUsed` should be an existing user.    
+Resource name `mailboxName` should not be empty, nor contain `% *` characters, nor starting with `#`.
+
+Response codes:
+
+- 200: The number unseen emails in a given mailbox
```
Comment (Arsnael) [inline src/site/markdown/server/manage-webadmin.md:1202]:
```suggestion
- 200: The number of unseen emails in a given mailbox
```

**Review by Arsnael — COMMENTED** [review]
Just some minor typo remarks but very good job :)

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you squash your fixups please?

**Comment by vttranlina** [comment]
Rebase master & Squash fixup

**Pull request merged**